virttest/bootstrap.py

```diff
@@ -208,19 +208,36 @@ def write_subtests_files(config_file_list, output_file_object, test_type=None):
         config_file.close()
 
 
-def get_directory_structure(rootdir, guest_file):
-    rootdir = rootdir.rstrip(os.sep)
-    start = rootdir.rfind(os.sep) + 1
+def guest_os_write_from_directory_structure(primary_dir,
+                                            output_file,
+                                            optional_dirs=[]):
+    """
+    Writes a `guest-os.cfg` file based on a number of config directories
+
+    :param primary_dir: the primary directory from where config files will
+                        be got and merged into output_file.
+    :param output_file: the file object that will receive writes with the
+                        content found on primary_dir and optional_dirs
+                        files.
+    :param optional_dirs: a list of optional directories, each one can contain
+                          contain the same directory structure as primary_dir.
+                          If a file is found in any of the optional_dirs, with
+                          the same relative path to the primary_dir (that is,
+                          same 'dir1/dir2/file.cfg', its contents will also be
+                          be added to output_file.
```

Comment:
@luckyh what you're proposing is basically a merge of directory (structure) contents, right?  I mean, right now `repo-b/foo.cfg` will only be added to the resulting file if `repo-a/foo.cfg` exists.

With the change you suggest, `repo-b/bar/foo.cfg` will have its content added even if `repo-a/bar/foo.cfg` doesn't exist, right?